moveit_ros/planning/planning_scene_monitor/src/current_state_monitor_rcl_interface.cpp

```
@@ -0,0 +1,81 @@
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Tyler Weaver */
+
+#include <moveit/planning_scene_monitor/current_state_monitor_rcl_interface.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <string>
+
+namespace planning_scene_monitor
+{
+namespace
+{
+static const auto SUBSCRIPTION_QOS = rclcpp::QoS(25);
+}
+
+CurrentStateMonitorRclInterface::CurrentStateMonitorRclInterface(const rclcpp::Node::SharedPtr& node) : node_(node)
+{
+}
+
+rclcpp::Time CurrentStateMonitorRclInterface::now() const
+{
+  return node_->now();
+}
+
+void CurrentStateMonitorRclInterface::createJointStateSubscription(const std::string& topic,
+                                                                   JointStateUpdateCallback callback)
+{
+  joint_state_subscription_ =
+      node_->create_subscription<sensor_msgs::msg::JointState>(topic, SUBSCRIPTION_QOS, callback);
+}
+
+void CurrentStateMonitorRclInterface::resetJointStateSubscription()
+{
+  joint_state_subscription_.reset();
+}
+
+std::string CurrentStateMonitorRclInterface::getJointStateTopicName() const
+{
+  if (joint_state_subscription_)
+  {
+    return joint_state_subscription_->get_topic_name();
+  }
+  else
+  {
+    return "";
```

should this also throw/print an error, if the string is undefined?

---

This is not a change in behavior (this is a straight copy-paste of code from current_state_monitor)